Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
-			startGold = startGold - startGold % 500;
-			pGameActivity->SetStartingGold(startGold);
+
+		if (gameActivity) {
+			int maxHumanPlayers = gameActivity->GetMaxPlayerSupport();
+			int minTeamsRequired = gameActivity->GetMinTeamsRequired();
+			if (humansInTeams > maxHumanPlayers) {
+				m_ScenarioButtons.at(ScenarioButtons::StartGameButton)->SetVisible(false);
+				std::string msgString = "Too many players assigned! Max for this activity is " + std::to_string(maxHumanPlayers);
+				m_StartErrorLabel->SetText(msgString);
+				m_StartErrorLabel->SetVisible(true);
+			} else if (minTeamsRequired > teamsWithPlayers) {
+				m_ScenarioButtons.at(ScenarioButtons::StartGameButton)->SetVisible(false);
+				std::string msgString = "Assign players to at\nleast " + std::to_string(minTeamsRequired) + " of the teams!";
+				m_StartErrorLabel->SetText(msgString);
+				m_StartErrorLabel->SetVisible(true);
+			} else if (teamWithHumans == 0) {
+				m_ScenarioButtons.at(ScenarioButtons::StartGameButton)->SetVisible(false);
+				m_StartErrorLabel->SetText("Assign human players\nto at least one team!");
+				m_StartErrorLabel->SetVisible(true);
+			} else {
+				m_ScenarioButtons.at(ScenarioButtons::StartGameButton)->SetVisible(true);
+				m_StartErrorLabel->SetVisible(false);
+			}
 		}
-    }
-
-	g_SceneMan.SetSceneToLoad(m_pSelectedScene, true, m_pDeployUnitsCheckbox->GetCheck());
-
-    // Set up the player and team assignments
-    pActivity->ClearPlayers(false);
-    for (int player = Players::PlayerOne; player < PLAYERCOLUMNCOUNT; ++player)
-    {
-        for (int team = Activity::TeamOne; team < TEAMROWCOUNT; ++team)
-        {
-            if (team != TEAM_DISABLED && m_aapPlayerBoxes[player][team]->GetDrawType() == GUICollectionBox::Image)
-            {
-                // Add the human players, not including CPU players
-                if (player != PLAYER_CPU)
-                    pActivity->AddPlayer(player, true, team, 0);
-                // CPU team, so mark it as such.. there are no actual CPU players
-                else if (pGameActivity)
-				{
-                    pGameActivity->SetCPUTeam(team);
-				}
-            }
-        }
-    }
-
-	for (int team = Activity::TeamOne; team < Activity::MaxTeamCount; ++team)
-    {
-		// Set up techs
-		GUIListPanel::Item *pTechItem = m_apTeamTechSelect[team]->GetSelectedItem();
-		if (pTechItem)
-		{
-			// If the "random" selection, choose one from the list of loaded techs
-			if (m_apTeamTechSelect[team]->GetSelectedIndex() == 1)//pTechItem->m_ExtraIndex < 0)
-			{
-				int selection = RandomNum<int>(1, m_apTeamTechSelect[team]->GetListPanel()->GetItemList()->size() - 1);
-				m_apTeamTechSelect[team]->SetSelectedIndex(selection);
-				pTechItem = m_apTeamTechSelect[team]->GetSelectedItem();
-
-				// Switch back to -Random-
-				m_apTeamTechSelect[team]->SetSelectedIndex(1);
+
+		char goldString[256];
+		if (m_GoldSlider->GetValue() == m_GoldSlider->GetMaximum()) {
```

Comment:
If you wanted to you could compact this so there's no if-else. It's probably fine as-is though.